### PR TITLE
Add ONNX LayerNormalization(17) 

### PR DESCRIPTION
**Pull request by skottmckay**
**Description**: LayerNormalization is now part of the ONNX spec as of opset 17.
We had a LayerNormalization contrib op, which (incorrectly) was registered in the ONNX domain. Use that implementation for the ONNX operator.

TODO: Check the ONNX spec matches the contrib op implementation to validate we can re-use it. 

Update skip_layer_norm_fusion.cc. There are other optimizers that use LayerNormalization that need updates as well.

**Motivation and Context**
#12916

### CODE REVIEW

**Comment by sunnyshu-intel** [comment]
- I have checked the changes suggested in this PR. I applied the changes on top of the main branch (Commit ID: 739b5675c8d28d3177ecc0f24cb82373014365f6), recompiled the code, and reinstall the wheel. 
- I am attaching the patch I applied on top of the main branch. 
PFA [12978.patch.txt](https://github.com/microsoft/onnxruntime/files/9579508/12978.patch.txt)

- Below is the script I used to check if this patch is working or not.

```python
# Reference : https://onnxruntime.ai/docs/performance/graph-optimizations.html
# Reference : https://onnxruntime.ai/docs/performance/mobile-performance-tuning.html
import onnxruntime as ort
import sys

def graph_optimization(input_model, output_model):

    sess_options = ort.SessionOptions()

    sess_options.graph_optimization_level = ort.GraphOptimizationLevel.ORT_ENABLE_EXTENDED

    sess_options.optimized_model_filepath = output_model

    #providers = kwargs.get("providers", ["CPUExecutionProvider"] if device.type == "cpu" else ["DnnlExecutionProvider"])

    session = ort.InferenceSession(input_model, sess_options, providers=["CPUExecutionProvider"])

    return session

if __name__ == "__main__":
    if len(sys.argv) != 3:
        print(f"[usage]: {sys.argv[0]} <input_model> <output_model>")
    else:
        graph_optimization(str(sys.argv[1]), str(sys.argv[2]))
```

- Attaching the input model **_(sln.onnx)_** for the script and the output model **_(sln_fused.onnx)_** generated by the script. 
PFA [sln_graph_optimization.zip](https://github.com/microsoft/onnxruntime/files/9579512/sln_graph_optimization.zip)

- The command I used to generate a model **_(sln_fused.onnx)_** consists of a SkipLayerNorm node fused from Add + LayerNorm from the input model **_(sln.onnx)_**.

```bash
python -u .\graph_optimization.py sln.onnx sln_fused.onnx
```

**Comment by wschin** [comment]
It has been a while so my memory could be wrong. When proposing layer normalization to ONNX, I reuse ORT's layer normalization with only minor change. Their sub-graphs are identical.
ONNX's layer normalization sub-graph:
https://github.com/wschin/onnx/blob/45eb3da54c54fb371016ea0902c0f8961a9286c1/onnx/defs/nn/defs.cc#L2642
ORT's layer normalization sub-graph:
https://github.com/microsoft/onnxruntime/blob/1a684152ccefab5b5c33e6bd66a16ef257940ef0/onnxruntime/core/graph/contrib_ops/contrib_defs.cc#L2178

File: onnxruntime/core/providers/cpu/cpu_execution_provider.cc [hunk 1]
```diff
@@ -758,6 +758,7 @@ class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 17, Ha
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 17, HannWindow);
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 17, MelWeightMatrix);
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 17, STFT);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 17, float, LayerNormalization);
```
Comment (yufenglee) [inline onnxruntime/core/providers/cpu/cpu_execution_provider.cc:761]: Why drop the double version?

Comment (skottmckay) [reply onnxruntime/core/providers/cpu/cpu_execution_provider.cc:789]: ONNX spec doesn't support double for Mean and InvStdDev inputs. The contrib op implementation requires matching types for all the inputs (based on the type constraints in the kernel registrations).

**Comment by sunnyshu-intel** [comment]
I have checked it with the latest [patch](https://patch-diff.githubusercontent.com/raw/microsoft/onnxruntime/pull/12978.patch). The method to check is described in this [reply](https://github.com/microsoft/onnxruntime/pull/12978#issuecomment-1248743483). It seems to work and give the desired results.

**Comment by sunnyshu-intel** [comment]
I have checked the new change and it seems to be working just fine and giving the desired results. No compilation error.

**Review by yufenglee — APPROVED** [review]

**Comment by yufenglee** [comment]
The PR LGTM. We also need to implement CUDA version.

**Pull request merged**